virttest/utils_net.py

```diff
@@ -1435,17 +1432,21 @@ def get_guest_ip_addr(session, mac_addr, os_type="linux", ip_version="ipv4",
             raise IPAddrGetError(mac_addr, "Unknown os type")
 
         if ip_version == "ipv4":
-            if nic_address["ipv4"]:
-                return nic_address["ipv4"][-1]
-            else:
-                return None
+            linklocal_prefix = "169.254"
+        elif ip_version == "ipv6":
+            linklocal_prefix = "fe80"
         else:
-            global_address = [x for x in nic_address["ipv6"]
-                              if not x.lower().startswith("fe80")]
-            if global_address:
-                return global_address[0]
+            raise IPAddrGetError(mac_addr, "Unknown ip version type")
```

Comment:
As the exception would be caught and raised again in line 1450~1452, I'd suggest to raise `ValueError` at here.